src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -558,23 +560,44 @@ def create_results(
     well_to_measurements: defaultdict[str, list[MeasurementData]] = defaultdict(
         list[MeasurementData]
     )
-    calculated_data: defaultdict[str, list[tuple[str, float]]] = defaultdict(
-        list[tuple[str, float]]
+    calculated_data: defaultdict[str, list[tuple[str, float | None]]] = defaultdict(
+        list[tuple[str, float | None]]
     )
     measurement_labels = [
         label for r_data in read_data for label in r_data.measurement_labels
     ]
+    error_documents_per_well: defaultdict[str, list[ErrorDocument]] = defaultdict(
+        list[ErrorDocument]
+    )
     for row_name, row in data.iterrows():
         label = str(row.iloc[-1])
         for col_index, value in enumerate(row.iloc[:-1]):
             well_pos = f"{row_name}{col_index + 1}"
-            well_value = try_non_nan_float_or_none(value)
-            # TODO: Report error documents for NaN values
-            if well_value is None:
+            well_value = try_float_or_none(value)
+            is_measurement = label in measurement_labels
+            # skip empty values
+            if isinstance(well_value, float) and math.isnan(well_value):
                 continue
-            if label in measurement_labels:
+
+            # Report error documents for NaN values
+            if well_value is None:
+                error_documents_per_well[well_pos].append(
+                    ErrorDocument(
+                        error=value,
+                        # TODO Add support for multiple read modes
+                        error_feature=read_data[0].read_mode.lower()
+                        if is_measurement
+                        else label,
+                    )
+                )
+            if is_measurement:
+
                 well_to_measurements[well_pos].append(
-                    MeasurementData(random_uuid_str(), well_value, label)
+                    MeasurementData(
+                        random_uuid_str(),
+                        NEGATIVE_ZERO if well_value is None else well_value,
+                        label,
+                    )
                 )
             else:
```

Comment:
```suggestion
            else if value is not None:
```

One last suggestion, put this here, and remove it from below